### PR TITLE
Logic dev lie mai rush hour

### PR DESCRIPTION
**Pull request by ItLieMai**
- Implement an explosion graphic when the enemy dies
- Make shuriken spin 
- Change the game's name in the main menu, from NinjaFurry to NinjaFury
- Check if time out and no enemy is left in the field,  Stage Clear

If you have any questions feel free to DM me.

### CODE REVIEW

File: src/application/logic/Explosion.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+
+import java.util.ArrayList;
+
+public class Explosion implements IRenderable {
+    private double x ,y ;
+    private int changedSprite = 0 ;
+    private long previousChange ;
+    protected boolean destroyed = false ;
```
Comment (Icegoo141) [inline src/application/logic/Explosion.java]: **IRenderable interface already has destroyed field**
- use this.destroyed instead
- set the initial value of fields in the constructor

File: src/application/logic/Explosion.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import application.sharedObject.RenderableHolder;
+import javafx.scene.canvas.GraphicsContext;
+import javafx.scene.image.Image;
+import javafx.scene.image.WritableImage;
+
+import java.util.ArrayList;
+
+public class Explosion implements IRenderable {
+    private double x ,y ;
+    private int changedSprite = 0 ;
```
Comment (Icegoo141) [inline src/application/logic/Explosion.java]: - set the value of this field in the constructor

File: src/application/logic/Explosion.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+            if (changedSprite >= 5) {
+                this.destroyed = true;
+            }
+            previousChange = now;
+        }
+    }
+
+    @Override
+    public void draw(GraphicsContext gc) {
+        int spriteX = changedSprite ;
```
Comment (Icegoo141) [inline src/application/logic/Explosion.java]: - unused variable

**Review by Icegoo141 — CHANGES_REQUESTED** [review]
fix the changes mentioned first

File: src/application/logic/Explosion.java [hunk 4]
```diff
@@ -37,21 +37,18 @@ public void update(){
         long now = GameController.getInstance().getGameLogic().getCurrTime();
         if (now - previousChange >= 12e7) {
             changedSprite++ ;
-            if (changedSprite >= 5) {
-                this.destroyed = true;
-            }
+            this.isDestroyed() ;
```
Comment (Icegoo141) [inline src/application/logic/Explosion.java]: - unused statement

**Review by Icegoo141 — CHANGES_REQUESTED** [review]
just 1 change left, go go

File: src/application/logic/Explosion.java (hunk 4 above)
Comment (ItLieMai) [reply src/application/logic/Explosion.java:40]: แก้ไปแล้วนะ ทำไมมันไม่ขึ้นนิ

**Review by Icegoo141 — APPROVED** [review]
ok

**Pull request merged**